ci/scripts/deterministic-recovery-test.sh

```diff
@@ -48,52 +49,48 @@ trap filter_stack_trace_for_all_logs ERR
 # NOTE(kwannoel): We must use `export` here, because the variables are not substituted
 # directly via bash subtitution. Instead, the `parallel` command substitutes the variables
 # from the environment. If they are declared without `export`, `parallel` can't read them from the env.
-export EXTRA_ARGS=""
-
-if [[ -n "${USE_SQL_BACKEND:-}" ]]; then
-  export EXTRA_ARGS="--sqlite-data-dir=."
-fi
+export EXTRA_ARGS="--sqlite-data-dir=."
```

Comment:
Can we default to use SQL backend with a temporary directory instead, so that we don't have to specify this?

Reply:
Also, I'm curious how we emulate SQLite? Will it really perform disk I/O? Can we simply use a in-memory instance that's accessible within the same process?

Reply:
Oh I think we can use in-memory instance indeed, now that we no longer kill meta-node (for now).

Reply:
> now that we no longer kill meta-node (for now)

In-memory database can also be shared by multiple lifespans of meta, as long as we keep holding a reference ourselves.

https://arc.net/l/quote/lrgjtbyn

Reply:
The reason why I kept this config is that I still prefer to use in-memory mode for simulation if it's not configured. But currently we can't, because we still have kill logic for meta in some integration tests. 🥵 
Besides the disk I/O indeed affected the duration of the test.

Reply:
> still prefer to use in-memory mode for simulation if it's not configured

By in-memory, do you mean in-memory kv? Can we switch to in-memory SQLite instead (also for `playground`)?

Reply:
> By in-memory, do you mean in-memory kv? Can we switch to in-memory SQLite instead (also for playground)?

I mean in-memory SQLite. Playground also uses it.